Allow PolygonSelector points to be removed

dstansby opened the pull request:
## PR Summary
This allows one to remove individual points of a `PolygonSelector` by right-clicking on them (mouse button 3).

## PR Checklist

<!-- Please mark any checkboxes that do not apply to this PR as [N/A]. -->

- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [x] New features are documented, with examples if plot related.
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [x] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [x] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

<!--
Thank you so much for your PR!  To help us review your contribution, please
consider the following points:

- A development guide is available at https://matplotlib.org/devdocs/devel/index.html.

- Help with git and github is available at
  https://matplotlib.org/devel/gitwash/development_workflow.html.

- Do not create the PR out of master, but out of a separate branch.

- The PR title should summarize the changes, for example "Raise ValueError on
  non-numeric input to set_xlim".  Avoid non-descriptive titles such as
  "Addresses issue #8576".

- The summary should provide at least 1-2 sentences describing the pull request
  in detail (Why is this change required?  What problem does it solve?) and
  link to any relevant issues.

- If you are contributing fixes to docstrings, please pay attention to
  http://matplotlib.org/devel/documenting_mpl.html#formatting.  In particular,
  note the difference between using single backquotes, double backquotes, and
  asterisks in the markup.

We understand that PRs can sometimes be overwhelming, especially as the
reviews start coming in.  Please let us know if the reviews are unclear or
the recommended next step seems overly demanding, if you would like help in
addressing a reviewer's comments, or if you have been waiting too long to hear
back on your PR.
-->

QuLogic commented on lib/matplotlib/widgets.py:
```diff
@@ -2794,6 +2800,34 @@ def __init__(self, ax, onselect, useblit=False,
+            self._xs.pop(i)
+            self._ys.pop(i)
+        if self._nverts <= 2:
+            # If only one point left, return to un-complete state to let user
```
```suggestion
            # If only one point left, return to incomplete state to let user
```

QuLogic commented on lib/matplotlib/widgets.py:
```diff
@@ -2794,6 +2800,34 @@ def __init__(self, ax, onselect, useblit=False,
+        if self._nverts <= 2:
+            # If only one point left, return to un-complete state to let user
+            # start drawing again
+            self._polygon_completed = False
```
Should add a test for this.

dstansby commented:
Thanks for the review. I've:
- Fixed typos
- Added a test for re-drawing the selector when only one point is left
- Added a note in the docstring that about the re-drawing behaviour

QuLogic commented on lib/matplotlib/tests/test_widgets.py:
```diff
@@ -500,3 +506,57 @@ def test_MultiCursor(horizOn, vertOn):
+    for (etype, event_args) in event_sequence:
+        do_event(tool, etype, **event_args)
+    # After removing two verts, only one remains, and the
+    # selector should be automatically resete
```
```suggestion
    # selector should be automatically reset.
```

dstansby commented:
@jklymak was this marked as a draft because it needed a rebase (now done), or is there something else?

jklymak commented:
Just the rebase.  

When looking at old PRs it is easy to hit "draft:false" in the search terms.  It is not easy to put "not:label" for the 10 labels that indicate author action needed.

dstansby commented:
👍, just checking

jklymak commented:
... when you clear the "needs rebase" totally feel free to move back to "Ready for Review".  Its just a  way to keep things off the radar until the author has cleared the issue.  Our PR backlog is still too big for this to be super effective, but I go through once a week or so and try and sort another dozen or so.

jklymak reviewed (approved):
I didn't test this, but looks good, and @QuLogic seemed to do a thorough review...
